### ISSUE TITLE
Mount `/` as Shared by default?

### ISSUE THREAD

**Issue by mangkoran** [microsoft/WSL#7477]
### Windows Build Number

Microsoft Windows [Version 10.0.22000.194]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.55-xanmod1-microsoft-standard-WSL2

### Distro Version

ArchWSL

### Other Software

- `podman` 3.3.1-1

### Repro Steps

1. Install `podman`
2. Configure `podman` to use rootful mode ([Reference](https://dev.to/bowmanjd/using-podman-on-windows-subsystem-for-linux-wsl-58ji))
2. Run any container, for convenience I exec `podman run docker.io/hello-world`

### Expected Behavior

Container run successfully without warning.

### Actual Behavior

Container run successfully with warning:
```
> podman run hello-world
WARN[0000] "/" is not a shared mount, this could cause issues or missing mounts with rootless containers
```

**Comment by Biswa96** [comment]
I think you are looking for cross distro mount which can be enabled in `/etc/wsl.conf` file by adding this

```ini
[automount]
crossDistro = true
```

More here https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-18970

**Comment by mangkoran** [comment]
Thank you, but I didn't found `/etc/wsl.conf` in my distro. Is it supposed to be there? Also I think [this release note](https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-17666) is also related to my issue.
> Add support for mount propagation flags [GH 2911].

**Comment by Biswa96** [comment]
That `/etc/wsl.conf` file has to be created manually.

**Comment by mangkoran** [comment]
I have created and added lines to `/etc/wsl.conf` and done `wsl --shutdown`, but it still showing `/` propagation is private:
```
~
> cat /etc/wsl.conf
[automount]
crossDistro = true


~
> findmnt / -o PROPAGATION,TARGET
PROPAGATION TARGET
private     /
```

**Comment by mangkoran** [comment]
I found [this](https://serverfault.com/questions/868682/implications-of-mount-make-private) on the web. Seems like the default mount propagation is indeed "private", but in `systemd`-ful OS `systemd` will change the mount propagation to "shared" ([Reference](https://github.com/systemd/systemd/blob/db9ecf050165fd1033c6f81485917e229c4be537/src/core/mount-setup.c#L528-L533)). I think we could use `bash` script to do this instead, but I'm not too proficient in shell scripting.

**Comment by KnairWang** [comment]
I find a way to solve this issue, by adding a boot command "`mount --make-rshared /`" into `wsl.conf`
(reference [Build 21286](https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-21286))

```
~
❯ podman ps
CONTAINER ID  IMAGE       COMMAND     CREATED     STATUS      PORTS       NAMES

~
❯ cat /etc/wsl.conf
[boot]
command="mount --make-rshared /"
```

I suppose after updating the `/etc/wsl.conf`, you may need run `wsl --shutdown` on Windows side to make the change work in WSL.

**Comment by mangkoran** [comment]
@KnairWang I have tried your solution (or workaround?) and it works!. Thank you very much. However, this solution (or workaround?) is might be the best we could have for `systemd`-less WSL distro. As WSL recently added native `systemd` support, I found that this issue still persist.

Closing this ticket and moving the issue for `systemd`-ful WSL distro to https://github.com/microsoft/WSL/issues/8922.
